### PR TITLE
fix lint-submodules errors

### PR DESCRIPTION
**Pull request by huangjiehua**
fix lint-submodules errors in#136

Signed-off-by: Jiehua Huang <huangjiehua@aisino.com>

### CODE REVIEW

File: test/integration/base_test_setup.go [hunk 1]
```diff
@@ -592,15 +592,24 @@ func GetKeyName(t *testing.T) string {
...
-	for _, key := range keys {
+	for _, k := range keys {
+		key := k
 		// Synchronous transaction
-		_, e := chClient.Execute(
-			channel.Request{
-				ChaincodeID: chaincodeID,
-				Fcn:         "invoke",
-				Args:        ExampleCCTxSetArgs(key, value),
+		_, e := retry.NewInvoker(retry.New(retry.TestRetryOpts)).Invoke(
```
Comment (bstasyszyn) [inline test/integration/base_test_setup.go:598]: I'm not sure I understand what you're trying to fix here? You basically have a retry invoker inside of a retry invoker.

File: test/integration/pkg/client/resmgmt/resmgmt_queries_pvt_test.go [hunk 2]
```diff
@@ -52,16 +54,33 @@ func TestQueryCollectionsConfig(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to create new resource management client: %s", err)
 	}
-
-	resp, err := client.QueryCollectionsConfig(orgChannelID, ccID, resmgmt.WithRetry(retry.DefaultResMgmtOpts))
+	resp, err := retry.NewInvoker(retry.New(retry.TestRetryOpts)).Invoke(
```
Comment (bstasyszyn) [inline test/integration/pkg/client/resmgmt/resmgmt_queries_pvt_test.go:57]: What does this fix?

**Review by bstasyszyn — COMMENTED** [review]
The title of the PR is to fix linter warnings but I see that you're trying to also fix something else - not sure what the problem was.

File: test/integration/base_test_setup.go (hunk 1 above)
Comment (huangjiehua) [reply test/integration/base_test_setup.go:598]: @bstasyszyn , I came into diffrent problems during integration tests, and these problems did not appear everytime. These problems as following:
`2020-08-05T14:17:44.8882001Z [35mintegration-tests_1  |[0m --- FAIL: TestLedgerQueries (14.27s)
2020-08-05T14:17:44.8882416Z [35mintegration-tests_1  |[0m     base_test_setup.go:604: 
2020-08-05T14:17:44.8882831Z [35mintegration-tests_1  |[0m         	Error Trace:	base_test_setup.go:604
2020-08-05T14:17:44.8883295Z [35mintegration-tests_1  |[0m         	            				channel_ledger_test.go:98
2020-08-05T14:17:44.8883742Z [35mintegration-tests_1  |[0m         	Error:      	Received unexpected error:
2020-08-05T14:17:44.8884705Z [35mintegration-tests_1  |[0m         	            	Chaincode status Code: (500) UNKNOWN. Description: error in simulation: failed to execute transaction 218fb6679801724fa0599626daf70f568e07b536219e1de6e06f9802748483fe: invalid invocation: chaincode 'example_cc_fabtest_00' has not been initialized for this version, must call as init first
2020-08-05T14:17:44.8885578Z [35mintegration-tests_1  |[0m         	            	Transaction processing for endorser [peer1.org1.example.com:7151]
2020-08-05T14:17:44.8886159Z [35mintegration-tests_1  |[0m         	            	github.com/hyperledger/fabric-sdk-go/pkg/fab/peer.(*peerEndorser).ProcessTransactionProposal
2020-08-05T14:17:44.8886722Z [35mintegration-tests_1  |[0m         	            		/opt/workspace/fabric-sdk-go/pkg/fab/peer/peerendorser.go:108
2020-08-05T14:17:44.8887275Z [35mintegration-tests_1  |[0m         	            	github.com/hyperledger/fabric-sdk-go/pkg/fab/peer.(*Peer).ProcessTransactionProposal
2020-08-05T14:17:44.8887779Z [35mintegration-tests_1  |[0m         	            		/opt/workspace/fabric-sdk-go/pkg/fab/peer/peer.go:216
2020-08-05T14:17:44.8888284Z [35mintegration-tests_1  |[0m         	            	github.com/hyperledger/fabric-sdk-go/pkg/fab/txn.SendProposal.func1
2020-08-05T14:17:44.8888777Z [35mintegration-tests_1  |[0m         	            		/opt/workspace/fabric-sdk-go/pkg/fab/txn/proposal.go:122
2020-08-05T14:17:44.8889199Z [35mintegration-tests_1  |[0m         	            	runtime.goexit
2020-08-05T14:17:44.8889620Z [35mintegration-tests_1  |[0m         	            		/usr/local/go/src/runtime/asm_amd64.s:1373
2020-08-05T14:17:44.8890302Z [35mintegration-tests_1  |[0m         	Test:       	TestLedgerQueries
2020-08-05T14:17:44.8890725Z [35mintegration-tests_1  |[0m         	Messages:   	Failed to reset keys
2020-08-05T14:17:44.8891059Z [35mintegration-tests_1  |[0m FAIL
2020-08-05T14:17:44.8891483Z [35mintegration-tests_1  |[0m FAIL	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/client/ledger	29.189s`
`2020-08-05T14:18:37.5246841Z [35mintegration-tests_1  |[0m --- FAIL: TestEventClient (0.01s)
2020-08-05T14:18:37.5247415Z [35mintegration-tests_1  |[0m     eventclient_test.go:46: Testing Deliver events
2020-08-05T14:18:37.5248042Z [35mintegration-tests_1  |[0m     --- FAIL: TestEventClient/Deliver_Filtered_Block_Events (0.01s)
2020-08-05T14:18:37.5249781Z [35mintegration-tests_1  |[0m         eventclient_test.go:142: CreateAndSendTransactionProposal return error: invoking transaction proposal failed: Transaction processing for endorser [peer1.org1.example.com:7151]: Chaincode status Code: (500) UNKNOWN. Description: error in simulation: failed to execute transaction 41355695064cd196c9e581f2e9ad7b8b4e3ad48829f96b270edf5d96a3304f64: invalid invocation: chaincode 'example_cc_fabtest_09lpjtle5lv' has not been initialized for this version, must call as init first
2020-08-05T14:18:37.5251028Z [35mintegration-tests_1  |[0m FAIL
2020-08-05T14:18:37.5251559Z [35mintegration-tests_1  |[0m FAIL	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/fab	13.926s
2020-08-05T14:18:54.0723116Z [35mintegration-tests_1  |[0m ok  	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/fabsdk/provider	14.683s
2020-08-05T14:19:39.8181667Z [35mintegration-tests_1  |[0m ok  	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/gateway	43.857s
2020-08-05T14:19:39.8182279Z [35mintegration-tests_1  |[0m ?   	github.com/hyperledger/fabric-sdk-go/test/integration/util/runner	[no test files]
2020-08-05T14:19:39.8182672Z [35mintegration-tests_1  |[0m FAIL`
And the reason I guess is the ledger consistency, so I add some retry.

Comment (bstasyszyn) [reply test/integration/base_test_setup.go:598]: OK, so maybe the inner retry can be removed, i.e. channel.WithRetry(retry.DefaultChannelOpts)?

Comment (huangjiehua) [reply test/integration/base_test_setup.go]: I think you are right, and I will add some retryCodes about lifecycle to ResMgmtDefaultRetryableCodes.

File: test/integration/pkg/client/resmgmt/resmgmt_queries_pvt_test.go (hunk 2 above)
Comment (huangjiehua) [reply test/integration/pkg/client/resmgmt/resmgmt_queries_pvt_test.go]: The log can be found in here[https://dev.azure.com/Hyperledger/Fabric-SDK-Go/_build/results?buildId=19076&view=results](url). So I add some retry too.

File: pkg/common/errors/retry/defaults.go [hunk 3]
```diff
@@ -115,6 +115,7 @@ var ResMgmtDefaultRetryableCodes = map[status.Group][]status.Code{
 		status.Code(pb.TxValidationCode_ENDORSEMENT_POLICY_FAILURE),
 		status.Code(pb.TxValidationCode_MVCC_READ_CONFLICT),
 		status.Code(pb.TxValidationCode_PHANTOM_READ_CONFLICT),
+		status.Code(pb.TxValidationCode_ENDORSEMENT_POLICY_FAILURE),
```
Comment (troyronda) [inline pkg/common/errors/retry/defaults.go:118]: Duplicate of L115?

**Review by bstasyszyn — APPROVED** [review]

**Review by troyronda — APPROVED** [review]

**Pull request merged**